Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -93,4 +126,44 @@ namespace AzToolsFramework::Prefab
         return instance.has_value() && (&instance->get() == &m_focusedInstance->get());
     }
 
+    const AZ::IO::Path& PrefabFocusHandler::GetPrefabFocusPath()
+    {
+        return m_instanceFocusPath;
+    }
+
+    const int PrefabFocusHandler::GetPrefabFocusPathLength()
+    {
+        return aznumeric_cast<int>(m_instanceFocusVector.size());
+    }
+
+    void PrefabFocusHandler::OnEntityStreamLoadSuccess()
+    {
+        // Focus on the root prefab (AZ::EntityId() will default to it)
+        FocusOnOwningPrefab(AZ::EntityId());
+    }
+
+    void PrefabFocusHandler::RefreshInstanceFocusList()
+    {
+        m_instanceFocusVector.clear();
+        m_instanceFocusPath.clear();
+
+        AZStd::list<InstanceOptionalReference> instanceFocusList;
+
+        // Use a support list to easily push front while traversing the prefab hierarchy
+        InstanceOptionalReference currentInstance = m_focusedInstance;
+        while (currentInstance.has_value())
+        {
+            instanceFocusList.push_front(currentInstance);
+
+            currentInstance = currentInstance->get().GetParentInstance();
+        }
+
+        // Populate internals using the support list
+        for (auto& instance : instanceFocusList)
+        {
+            m_instanceFocusPath.Append(instance->get().GetContainerEntity()->get().GetName());
+            m_instanceFocusVector.emplace_back(instance);
+        }
+    }
```

Comment:
question: I'd be interested to see how this performs with a very large prefab (what's an common size we might expect? might be worth trying a worse case too?)

Reply:
Jeremy sent me all the info about setting up Pix, so I'm going to be setting that up on Monday. Also, Nick is working on setting up a Python test to do performance checks on the Editor, which should be useful to gauge the impact of changes at every step.